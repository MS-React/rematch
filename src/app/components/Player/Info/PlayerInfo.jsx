import React from 'react';
import PropTypes from 'prop-types';

const InfoPlayer = ({ name }) => (
  <div key={'player-info'} className="playerInfo">
    Hello! {name}
  </div>
);

InfoPlayer.propTypes = {
  player: PropTypes.shape({
    name: PropTypes.string
  }).isRequired
};

export default InfoPlayer;
