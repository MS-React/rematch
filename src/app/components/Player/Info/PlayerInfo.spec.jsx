import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PlayerInfo from './PlayerInfo';

describe('PlayerInfo', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      player: {
        name: 'John Doe'
      }
    };
    wrapper = shallow(<PlayerInfo {...defaultProps} />);
  });

  it('should show the user an input to choose his nickname an a button to start', () => {
    expect(renderer.create(wrapper).toJSON())
      .toMatchSnapshot();
  });
});
