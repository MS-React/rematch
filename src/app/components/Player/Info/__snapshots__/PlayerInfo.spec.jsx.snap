// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfo should show the user an input to choose his nickname an a button to start 1`] = `
<div
  className="playerInfo"
>
  Hello! 
</div>
`;
